src/lib.rs

```diff
@@ -164,9 +164,11 @@ pub trait Die<T> {
     /// x.die_code("strange", 3); // prints `strange` to stderr then exits with code 3
     /// ```
     fn die_code(self, msg: &str, exit_code: i32) -> T;
+
+    fn die_with<X: PrintExit>(self, func: fn(E) -> X) -> T;
```

Comment:
If you add the following test to `tests/integration_test.rs`, you'll notice that it doesn't compile:
```rust
#[test]
#[should_panic]
fn die_with_capture() {
    let code = 3;
    Result::<(), &str>::Err("error text").die_with(|error| (format!("Error is: {}", error), code));
}
```
This is because `die_with` expects a function pointer (`fn`) and therefore cannot accept closures with captures.

Instead of using a function pointer, you should generally prefer using the function traits. In this case, I would use the `FnOnce` trait. That way, you can pass closures that capture from their environment.